core/src/validator.rs

```diff
@@ -2056,28 +2056,26 @@ fn move_and_async_delete_path(path: impl AsRef<Path> + Copy) {
     ));
 
     if path_delete.exists() {
-        debug!("{} exists, delete it first.", path_delete.display());
         std::fs::remove_dir_all(&path_delete).unwrap();
     }
 
     if !path.as_ref().exists() {
-        info!(
-            "move_and_async_delete_path: path {} does not exist",
-            path.as_ref().display()
-        );
         return;
     }
 
-    std::fs::rename(&path, &path_delete).unwrap();
+    if let Err(err) = std::fs::rename(&path, &path_delete) {
+        warn!(
+            "Path renaming failed: {}.  Falling back to rm_dir in sync mode",
+            err.to_string()
+        );
+        std::fs::remove_dir_all(&path).unwrap();
+        return;
```

Comment:
seems like we can easily avoid this early return by using an `} else {`

Reply:
I usually let the the function do the early return for various exceptions, and let the later main part of the function represent the more likely flow, and avoid too much indention in that part.  I can make the change if less early return is really preferred.

Reply:
My pref in this case would be to remove the early return but I'm not going to attempt to force compliance.